unittests/test_cli.py

```diff
@@ -156,30 +157,25 @@ def test_check_retry_failed(run_reframe, tmp_path, logfile):
     returncode, stdout, _ = run_reframe(
         checkpath=['unittests/resources/checks_unlisted/deps_complex.py'],
     )
-    assert 'T0' in stdout
-    assert 'T1' in stdout
-    assert 'T2' in stdout
-    assert 'T3' in stdout
-    assert 'T4' in stdout
-    assert 'T5' in stdout
-    assert 'T6' in stdout
-    assert 'T7' in stdout
-    assert 'T8' in stdout
-    assert 'T9' in stdout
     returncode, stdout, _ = run_reframe(
         checkpath=['unittests/resources/checks_unlisted/deps_complex.py'],
         more_options=['--retry-failed', f'{tmp_path}/report.json']
     )
-    assert 'T0' not in stdout
-    assert 'T1' not in stdout
-    assert 'T2' in stdout
-    assert 'T3' not in stdout
-    assert 'T4' not in stdout
-    assert 'T5' not in stdout
-    assert 'T6' not in stdout
-    assert 'T7' in stdout
-    assert 'T8' in stdout
-    assert 'T9' in stdout
+    with open(f'{tmp_path}/report.json') as fp:
+        report = json.load(fp)
+
+    report_summary = {t['name']: t['fail_phase']
```

Comment:
```suggestion
    failing_phase = {t['name']: t['fail_phase']
```

Also the lines below are not correctly aligned, that's why PEP8 complains I think.